File: LeetCode75/Level1/Is_Subsequence/Is_Subsequence.py
# 392. Group Anagrams Python
class Solution(object):
    def isSubsequence(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """
        pre = -1
        for c in s:
            if c not in t: return False
            index = t.find(c, pre+1)
            if index == -1: return False
            pre = index
        
        return True

File: LeetCode75/Is_Subsequence/Is.Subsequence.cc
// 392. Group Anagrams C++
#include <string>

class Solution {
public:
    bool isSubsequence(std::string s, std::string t) {
        int len_s = s.size();
        int len_t = t.size();
        int pre = -1;
        bool exist = false;
        for (int i = 0; i < len_s; i++) {
            for (int j = pre + 1; j < len_t; j++) {
                if (t[j] == s[i]) {
                    pre = j;
                    exist = true;
                    break;
                }
            }
            if (!exist) return false;
            exist = false;
        }
        return true;
    }
};
